pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -51,7 +51,7 @@ var (
 	insufficientPrivilegesRegexp = regexp.MustCompile(`(?i)(AccessDenied|Forbidden|deny|denied)`)
 	dependenciesRegexp           = regexp.MustCompile(`(?i)(PendingVerification|Access Not Configured|accessNotConfigured|DependencyViolation|OptInRequired|DeleteConflict|Conflict|inactive billing state|ReadOnlyDisabledSubscription|is already being used|InUseSubnetCannotBeDeleted|VnetInUse)`)
 	resourcesDepletedRegexp      = regexp.MustCompile(`(?i)(not available in the current hardware cluster|InsufficientInstanceCapacity|SkuNotAvailable|ZonalAllocationFailed)`)
-	configurationProblemRegexp   = regexp.MustCompile(`(?i)(AzureBastionSubnet|not supported in your requested Availability Zone|InvalidParameterValue|notFound|NetcfgInvalidSubnet|InvalidSubnet|Invalid value)`)
+	configurationProblemRegexp   = regexp.MustCompile(`(?i)(AzureBastionSubnet|not supported in your requested Availability Zone|InvalidParameterValue|notFound|NetcfgInvalidSubnet|InvalidSubnet|Invalid value|KubeletHasInsufficientMemory|KubeletHasDiskPressure|KubeletHasInsufficientPID)`)
```

Comment:
Mhm, why is KubeletHasInsufficientMemory necessarely a (shoot) - configuration problem? To my knowledge that can temporarely happen if the node has pods scheduled with burstable or best-effort QoS class (c-groups do not have memory limit and can therefore overshoot).  I think that is more a workload problem rather than a Shoot configuration issue  like suggested here.

Reply:
The name "configuration problem" does not specify where/what the configuration problem is. It's an "umbrella" code that can be used in multiple situations.
The rationale behind this PR is to help operators that work with the Dashboard and the list of "clusters with issues" by hiding shoots where an operator cannot help anyways. If the kubelet has insufficient memory, for example, then the operator can't do anything. As you said, it's a workload (resources requests/limits configuration) problem and out of the operator's control. As the Dashboard/Gardener is using error codes to indicate user/infrastructure problems I think this code is a good fit for such kind of situations.

Reply:
>   It's an "umbrella" code that can be used in multiple situations.
Yes the operator does not care if it is a Shoot configuration problem or a workload related problem. 
However not only operators but also end-users are typically using the Gardener dashboard / check the Shoot conditions to check for the health of their Shoot. For them I think it is helpful to differentiate between these two cases. 
Shouldn't be the  dashboard independent of the Gardener? So the Dashboard could just filter both of these cases for the operator - but at least the Shoot owner could differentiate.

Reply:
Why is "configuration problem" not good enough? It's just an approximation of the care controller to categorize errors. In order to understand more/the concrete error is not part of the error code but of the condition reason/message. I don't see the necessity to introduce a second error code.

Reply:
@danielfoehrKn are you okay with proceeding for now? Does my argumentation make sense?

Reply:
yes, go ahead.